waltid-crypto/src/jvmMain/kotlin/id/walt/crypto/keys/oci/OCIKey.jvm.kt

```diff
@@ -286,3 +296,31 @@ actual class OCIKey actual constructor(
 }
 
 
+@ExperimentalTime
+private suspend fun <T> retry(
+    maxDuration: Duration = Duration.ofSeconds(2),
+    retryInterval: Duration = Duration.ofMillis(100),
+    block: suspend () -> T
+): T {
+    var result: Result<T>
+    var totalDuration = Duration.ZERO
+
+    while (totalDuration < maxDuration) {
+        val elapsedTime = measureTime {
```

Comment:
hm..I was thinking about adding up time waited for, but this also works
since the execution time is also part of it, maybe worth specifying a bigger number as maxDuration?